### PR TITLE
v1.5

### LINKED ISSUE #81: v1.5
Design: https://www.figma.com/file/6tujIX9FtkIAUSjYycJUHi/Alephium---Website-ecosystem?node-id=0%3A1
Content: https://github.com/AlephiumFoundation/Website/issues/10#issuecomment-1065025098

Notes:
* Omit the usability section
* Static design for this iteration (no parallax, no confetti)

### PR DESCRIPTION
**Pull request by lf94**
Resolves https://github.com/alephium/www/issues/81

### CODE REVIEW

**Comment by lf94** [comment]
The test is failing on a positive: that link, when clicked from the test, is not 404... We can ignore it in `.github/broken-link-check-config.json` or not use the link? Not sure what to do.

**Review by nop33 — CHANGES_REQUESTED** [review]
Good job @lf94! I hope you don't take my review very critically, this is what I do best, so I am very critical about implementing designs 😇  Code review will follow once I am happy with visual, functional, and performance stuff.

## Visual and functional review
1. The night sky background does not expand to the whole width of the screen and it creates a sharp cut
image
1. The team and career links are not working properly
    image
1. All font sizes and image dimensions all across the website are not accurate with the designs. Example:
image
1. In the designs, the cards take all available content space (full content width). In the implementation the are aligned to the left 
    image
1. Confetti background is missing in milestones section
1. The year is not sticky when scrolling (see [here for example](https://www.simoninstitute.ch/project/longterm-governance-network/))
1. Rows don't expand to full width of screen, like in figma 
    image
1. testnet launch is not highlighted 
    image
1. More spacing is needed between the title and the date (probably a `line-height` issue) 
    image
1. Not enough spacing between sections 
    image
1. Cards have unequal size: 
    image
1. Mobile 
    image 
    image

## Performance review
1. Performance slightly worsened  
    Live website 
    image
    
    PR code after building and serving with `npx http-server`: 
    image
2. Console errors reduce best practices score (CSP needs to be updated)

## Comments about things not present in Figma
1. Technology page is not accessible from anywhere
1. Technology page has no navigation menu

## Questions
1. Is usability section now gone for good?
1. Technology section is still present even though there's a subpage for it, is this intended?

**Comment by lf94** [comment]
Would you happen to know how to export the night sky background from Figma without it cutting off? I thought originally it too was the element clipping it, but it's the actual SVG...!

**Comment by nop33** [comment]
> Would you happen to know how to export the night sky background from Figma without it cutting off? I thought originally it too was the element clipping it, but it's the actual SVG...!

Normally, through the Figma app, @mvaivre has given me "edit" rights, which allows me to properly export assets (even if they are cut-off). @mvaivre, could you add the website project on Figma as well, so that I can export what @lf94 needs?

**Comment by mvaivre** [comment]
First of all, thanks @lf94 for the work and @nop33 for the thorough hands-on review!

> Confetti background is missing in milestones section

We discussed this with Lee and it's normal. This was a "placeholder" for a potential animation I'm not so sure about. So we start without it 👌 

> could you add the website project on Figma as well

Done!

**Comment by mvaivre** [comment]
> Is usability section now gone for good?

Unfortunately, it's gone for now, as the marketing team feared that this would make the page too long. I'm hopeful to get it back as a standalone download page though, because it's nice.

> Technology section is still present even though there's a subpage for it, is this intended?

The marketing team decided to have a (slightly) more condensed tech section on the landing page (with no tips boxes & "see more" links), plus a standalone page reusing the previous content. I'm personally not 100% convinced by this approach, but we can iterate easily if needed. Let's see how this looks.

**Comment by nop33** [comment]
> Would you happen to know how to export the night sky background from Figma without it cutting off? I thought originally it too was the element clipping it, but it's the actual SVG...!

Here are the assets you need. I provide you with 3 different ones: sky, eclipse, and one that combines them both. Idk how you prefer implementing it, feel free to choose!

<details><summary>Click here to see assets</summary>

## stars
stars

## eclipse
eclipse

## eclipse with stars
eclipse-with-stars

</details>

**Comment by LeeAlephium** [comment]
1-5 is addressed (either by Mika or via code), 7-12 is addressed (same), CSP issue is addressed (so best practices score should be back up)

Technology page is still not accessible; I haven't been told how we should go there yet (nor how people will get back to the home page from there, as you noted)... We can do additional PRs though once things are figured out for this piece...

The only thing which really remains is 6 (sticky years). Even re-arranging the DOM does not help me - I think it has to live way too high up the DOM to which it won't be styled correctly if it is. I really don't know what to do.

All right then, let's do another round...! The long format you're doing is working well :)

**Comment by nop33** [comment]
> All right then, let's do another round...!

Is this a request for review? Would be useful if you would press the "re-request review" button in such cases! (no need for this time) Thanks!

image

File: src/styles/themes.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ import { DefaultTheme } from 'styled-components'
 
 export type ThemeType = 'light' | 'dark'
 
+export const oppositeThemeOf = (theme) => (theme === darkTheme ? lightTheme : darkTheme)
```
Comment (mvaivre) [inline src/styles/themes.ts:5]: This causes an editor error linked to type: `any`.

image


Also, this is not used anywhere, so what should we do with it? :)

File: src/styles/themes.ts [hunk 2]
```diff
@@ -10,9 +12,9 @@ export const darkTheme: DefaultTheme = {
   textSecondary: 'var(--color-grey-200)',
   textTertiary: 'var(--color-grey-200)',
 
-  bgPrimary: 'var(--color-black)',
+  bgPrimary: 'var(--color-grey-900)',
```
Comment (mvaivre) [inline src/styles/themes.ts:15]: If 2 theme variables end up having the same value, it's probably that we don't use them properly. I'll see how to streamline this situation!

File: src/styles/themes.ts (hunk 1 above)
Comment (mvaivre) [reply src/styles/themes.ts:5]: Actually, seems like your local typescript compiler isn't screaming at you anymore? Mine is upset :D 

image

File: src/components/DualTimeline.tsx [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+  )
+
+Year = styled(Year)`
+  position: sticky;
+  top: 0;
+  background: white;
+  z-index: -1;
+`
+
+const LineStartsEnds = ({ headings, starts, ends, isSingle, height }) =>
```
Comment (mvaivre) [inline src/components/DualTimeline.tsx:268]: No types, conditional props?

File: src/components/DualTimeline.tsx [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+`
+
+const YearData = styled.div`
+  width: 100%;
+  padding-bottom: 3rem;
+  font-weight: var(--fontWeight-semiBold);
+  font-size: var(--fontSize-28);
+`
+
+let Year = ({ value, headings, isSingle }) =>
```
Comment (mvaivre) [inline src/components/DualTimeline.tsx:237]: We're mixing components and styles. Our internal rule so far, was to keep all stylings at the end of the file.

File: package.json [hunk 5]
```diff
@@ -21,9 +21,12 @@
     "lint:fix": "eslint . --fix --ext .js,.jsx,.ts,.tsx"
   },
   "dependencies": {
+    "@alephium/sdk": "^0.0.6",
```
Comment (nop33) [inline package.json:24]:
```suggestion
    "@alephium/sdk": "0.0.6",
```

File: src/components/DualTimeline.tsx [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+
+const Container = styled.div`
+  width: 100%;
+  margin-bottom: 226px;
+`
+
+const Text = styled.div`
+  ${({ isMajor }) =>
+    isMajor &&
+    `
```
Comment (nop33) [inline src/components/DualTimeline.tsx:169]: Although this is correct, using `css` would improve the code highlighting
```suggestion
    css`
```

File: src/utils/index.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  const denominator = maxDecimals === 1 ? 10 : 100
+  const preciseNumber = (Math.round((formatedNumber + Number.EPSILON) * denominator) / denominator).toString()
+  const numberParts = preciseNumber.split('.')
+  const numberInteger = numberParts[0]
+  const numberDecimal = (numberParts.length > 1 && `.${numberParts[1]}`) || undefined
+
+  return [numberInteger, numberDecimal, suffixes[suffixIndex]]
+}
```
Comment (nop33) [inline src/utils/index.tsx:25]: Did we implement something like this for the stats in the explorer? Maybe a good candidate for the js-sdk?

File: src/components/PageSectionNumbers.tsx [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    },
+    {
+      value: `${addApostrophes(hashrateInteger)}${hashrateDecimal ?? ''}`,
+      isLoading: false,
+      description: `${hashrateSuffix}H/s`
+    },
+    {
+      value: supply,
+      isLoading: false,
+      description: 'alph circulating'
```
Comment (nop33) [inline src/components/PageSectionNumbers.tsx:103]: Looks weird with lower case:

image

```suggestion
      description: 'ALPH circulating'
```

**Comment by nop33** [comment]
image

**Comment by mvaivre** [comment]
I think we can close this one, as it's superseded by #83. Thanks for your work!

**Pull request closed without merging**